runtime/rasdump/dmpmap.c

```diff
@@ -121,7 +127,33 @@ static const DgDumpEnvSwitch dgSwitches[] = {
 			}
 };
 
-static const int numDumpEnvSwitches = ( sizeof(dgSwitches) / sizeof(DgDumpEnvSwitch) );
+typedef struct OomDefaultSetting
+{
+	const char *typeString;
+	const char *onOpt;
+} OomDefaultSetting;
+
+static const OomDefaultSetting oomDefaultTable[] = {
+		{
+				HEAP_KIND,
+				SYSTHROW_OOM_1_4
+		},
+		{
+				JAVA_KIND,
+				SYSTHROW_OOM_1_4
+		},
+		{
+				SNAP_KIND,
+				SYSTHROW_OOM_1_4
+		},
+		{
+				SYSTEM_KIND,
+				"events=systhrow,range=1..1,filter=java/lang/OutOfMemoryError,request=exclusive+compact+prepwalk"
+		}
+};
+
+static const int numDumpEnvSwitches = sizeof(dgSwitches) / sizeof(dgSwitches[0]);
+static const int numOomSettings = sizeof(oomDefaultTable) / sizeof(oomDefaultTable[0]);
```

Comment:
The type of these constants should be `UDATA` instead of `int`; as is we have signed-unsigned comparisons which are likely to trigger (potentially fatal) warnings.

Reply:
Will do.  Note that `static const int numDumpEnvSwitches` is legacy code, so we know it compiles.

Reply:
We don't know that: The comparison involving `numDumpEnvSwitches` is with `IDATA` so no signed-unsigned mismatch.